### PR TITLE
🏗 Copy form values from the browser to the HTML snapshot that gets uploaded to Percy for visual-diff tests

### LINKED ISSUE #20630: Visual testing framework: hard code input values into DOM snapshot to send to Percy
Currently values in input fields get lost when the DOM is snapshotted and sent to Percy, so there's no way to verify that values are displaying correctly (especially after actions).

Ideally we would find any input fields, read their values, and then add a `value={someValue}` attribute to them so they'll be displayed in Percy.

Even more ideally, there'd be a way to check for input values in our puppeteer scripts. So having a `verifyInputValue` counterpart to `verifyCssElements`

cc @danielrozenberg

### PR DESCRIPTION
**Pull request by danielrozenberg**
This PR:

* Fixes #20630
* Adds an error handler to display what went wrong when `require`ing an interactive test file crashes the visual diff tests runner
* Adds sample `<amp-form>` visual diff tests to show that this really fixes #20630

### CODE REVIEW

File: build-system/tasks/visual-diff/snippets/freeze-form-values.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+// This file is executed via Puppeteer's page.evaluate on a document to remove
+// all <script> tags that import AMP pages. This makes for cleaner diffs and
+// prevents "double-execution" of AMP scripts when enableJavaScript=true.
```
Comment (rsimha) [inline build-system/tasks/visual-diff/snippets/freeze-form-values.js]: Nice. Double execution was an annoying problem when we used `enableJavaScript=true` a long time ago. Does this PR re-enable that option?

**Comment by andrewwatterson** [comment]
Bammo! Thanks. :)

**Comment by rsimha** [comment]
New visual tests approved. Nice work!

File: build-system/tasks/visual-diff/snippets/freeze-form-values.js (hunk 1 above)
Comment (danielrozenberg) [reply build-system/tasks/visual-diff/snippets/freeze-form-values.js]: Oops, this comment comes from the file `remove-amp-scripts.js`, which I had copied to create this file. Pushed an updated comment.

The fix for `enableJavaScript=true` has been in for a while now :)

**Pull request merged**